tempodb/backend/readerat.go

```diff
@@ -41,7 +41,7 @@ func (b *backendReader) ReadAt(ctx context.Context, p []byte, off int64) (int, e
 
 // ReadAll implements ContextReader
 func (b *backendReader) ReadAll(ctx context.Context) ([]byte, error) {
-	return b.r.Read(ctx, b.name, b.meta.BlockID, b.meta.TenantID)
+	return b.r.Read(ctx, b.name, b.meta.BlockID, b.meta.TenantID, ShouldCache(b.name))
```

Comment:
@joe-elliott  fyi - ContextReader still needs `shouldCache()` unless we modify its initialisation function to 
```
// NewContextReader creates a ReaderAt for the given BlockMeta
func NewContextReader(meta *BlockMeta, name string, r Reader, shouldCache bool) ContextReader

```

Reply:
^ made that change.